e2e2/test/images/neuron/e2e-test/bert-infer.py

```diff
@@ -0,0 +1,99 @@
+import os
+import time
+import torch
+import torch_neuronx
+from transformers import BertForPreTraining, BertTokenizer
+from torch.utils.data import DataLoader, TensorDataset
+
+
+def create_dummy_data(tokenizer, num_samples=104, max_length=128):
+    # Create dummy input data
+    sentences = [
+        "This is a dummy sentence number {}".format(i) for i in range(num_samples)
+    ]
+    tokenized_inputs = tokenizer(
+        sentences,
+        max_length=max_length,
+        padding="max_length",
+        truncation=True,
+        return_tensors="pt",
+    )
+    labels = tokenized_inputs.input_ids.detach().clone()
+
+    # MLM task: randomly mask some tokens
+    mlm_probability = 0.15
+    input_ids, labels = mask_tokens(
+        tokenized_inputs.input_ids, tokenizer, mlm_probability
+    )
+
+    return TensorDataset(
+        input_ids, tokenized_inputs.attention_mask
+    )
+
+
+def mask_tokens(inputs, tokenizer, mlm_probability):
+    labels = inputs.clone()
+    probability_matrix = torch.full(labels.shape, mlm_probability)
+    special_tokens_mask = [
+        tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True)
+        for val in labels.tolist()
+    ]
+    probability_matrix.masked_fill_(
+        torch.tensor(special_tokens_mask, dtype=torch.bool), value=0.0
+    )
+    masked_indices = torch.bernoulli(probability_matrix).bool()
+    labels[~masked_indices] = -100  # We only compute loss on masked tokens
+
+    inputs[masked_indices] = tokenizer.convert_tokens_to_ids(tokenizer.mask_token)
+
+    return inputs, labels
+
+
+def run_inference(model, tokenizer, batch_size, mode):
+    model.eval()
+
+    dataset = create_dummy_data(tokenizer)
+    dataloader = DataLoader(dataset, batch_size=batch_size)
+
+    total_time = 0
+    total_batches = len(dataloader)
+    b = next(iter(dataloader))
+    inputs, masks = b
+
+    model_neuron = torch_neuronx.trace(model, (inputs, masks))
+    # # Compile the model to neuron device
+    filename = 'model.pt'
```

Comment:
This is just a general comment, nothing for you to do here. But, seeing this, it makes me thing it would probably be a good idea for us to manage our own model artifacts, and not rely on pulling them from Huggingface. We'll have to consider where we want them stored, maybe multiple regions, and make sure they're well documented and including metadata in case we ever start adding a large number of models.